### PR TITLE
feat(web): Add script to deploy KeymanWeb release to s.keyman.com

### LINKED ISSUE #4239: bug(web): release build should create a PR for changes to s.keyman.com
**Describe the bug**

Currently, the deployment to s.keyman.com pushes directly to master. We should instead be creating a branch, pushing to that branch, opening a PR on this and merging it automatically. This gives us a pathway for when the build fails to deploy (e.g. two pushes to s.keyman.com at the same time).

Furthermore, the build currently does not fail out when the git push fails. It needs to!

This is part of step 5 in the keymanweb build at https://build.palaso.org/admin/editBuildRunners.html?id=buildType:Keymanweb_Build

### PR DESCRIPTION
**Pull request by darcywong00**
Fixes #4239

Infrastructure effort to deploy KeymanWeb release to s.keyman.com via a GitHub branch.
~This PR also changes KeymanWeb deployment to s.keyman.com/kmw/engine/$VERSION/**src**/~

Note: sourcemap path still needs to be resolved in #5149

### CODE REVIEW

File: web/s_keyman_com.sh [hunk 1]
```diff
@@ -0,0 +1,154 @@
+#!/bin/bash
```
Comment (jahorton) [inline web/s_keyman_com.sh:1]: Could we put this script within a subfolder?

For reference, the iOS platform now has a CI script at `ios/tools`.

File: web/s_keyman_com.sh [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+##
+## Remap sourcemap paths
+##
+function remap_sourcemap_paths {
+  local dstpath="$S_KEYMAN_COM/kmw/engine/$VERSION/src"
+
+  cd $dstpath
+  for mapfile in *.map; do
+    echo "Remapping sourcempath paths for: $mapfile"
```
Comment (jahorton) [inline web/s_keyman_com.sh:98]:
```suggestion
    echo "Remapping sourcemap paths for: $mapfile"
```

**Review by jahorton — COMMENTED** [review]
> This PR also changes KeymanWeb deployment to s.keyman.com/kmw/engine/$VERSION/src/

As opposed to...?  We should probably conform to the existing pattern if possible.

Note:  I plan to leave the final review to @mcdurdin, since the script does kind of expose how the artifacts get uploaded.  He's got a better mind for potential security issues than I.

**Comment by darcywong00** [comment]
> As opposed to...? We should probably conform to the existing pattern if possible.

@mcdurdin requested a trailing `src/` folder in the s.keyman.com path

Currently, the artifacts are in the `$VERSION/` like [15.0.23](https://github.com/keymanapp/s.keyman.com/tree/master/kmw/engine/15.0.53)

File: web/s_keyman_com.sh (hunk 1 above)
Comment (darcywong00) [reply web/s_keyman_com.sh:1]: Possibly. Though `web/tools/` currently has recorder test pages

**Comment by mcdurdin** [comment]
> @mcdurdin requested a trailing `src/` folder in the s.keyman.com path

That's only for the source files, not the compiled files, and matches what we currently have. See for example https://github.com/keymanapp/s.keyman.com/tree/master/kmw/engine/14.0.274

**Comment by jahorton** [comment]
Is this intended to address #4239?  It's not currently tagged.

**Comment by darcywong00** [comment]
> Is this intended to address #4239? It's not currently tagged.

Oh right. I'll update the PR description to tag this. The motivation for the change was to replace the CI Powershell scripts so they can run on the Linux build agent.

**Review by mcdurdin — APPROVED** [review]
LGTM; just consider moving script as Joshua suggested?

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.58-alpha](https://keyman.com/downloads/releases/alpha/15.0.58)

**Pull request merged**